pkg/flowkit/util/mixpanel.go

```diff
@@ -19,44 +19,41 @@
 package util
 
 import (
-	"encoding/json"
-	"io/ioutil"
-	"net/http"
+	"github.com/spf13/cobra"
 	"strings"
 )
 
 const (
-	MIXPANEL_TRACK_URL = "https://api.mixpanel.com/track"
+	MIXPANEL_EVENT_PROJECT_TOKEN = "token"
+	MIXPANEL_EVENT_CALLER        = "caller"
+
+	FLOW_CLI               = "flow-cli"
+	MIXPANEL_PROJECT_TOKEN = "da53727e0435e12820c831098691f8e5"
 )
 
-type MixpanelClient struct {
-	token   string
-	baseUrl string
+type event struct {
+	Name       string                 `json:"event"`
+	Properties map[string]interface{} `json:"properties"`
 }
 
-func SendEvent(command string) error {
-	mixpanelEvent := NewEvent(command)
-	mixpanelEvent.SetUpEvent(MIXPANEL_PROJECT_TOKEN, FLOW_CLI)
-	eventPayload, _ := encodePayload(mixpanelEvent)
-	payload := strings.NewReader(eventPayload)
-	req, _ := http.NewRequest("POST", MIXPANEL_TRACK_URL, payload)
-	req.Header.Add("Accept", "text/plain")
-	req.Header.Add("Content-Type", "application/json")
-	res, _ := http.DefaultClient.Do(req)
-
-	defer res.Body.Close()
-	_, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return err
+func newEvent(command *cobra.Command) *event {
+	return &event{
+		processCommandName(command),
+		make(map[string]interface{}),
 	}
-	return nil
-
 }
-func encodePayload(obj interface{}) (string, error) {
-	b, err := json.Marshal(obj)
-	if err != nil {
-		return "", err
+
+//transform command path name e.g. flow accounts get -> accounts-get
+func processCommandName(command *cobra.Command) string {
```

Comment:
I feel it would be nicer if we get the whole command as it is eg: `flow accounts get` than adding the dashes, because some of the commands have dashes in names like: `flow transactions send-signed` which then becomes `transactions-send-signed` and it's not clear what it is. The only thing worth modifying is stripping the first `flow`, but even that I don't mind. Also why are you checking if first command is `flow` is that ever not true?

Reply:
addressed